### PR TITLE
Add more dartdocs

### PR DESCRIPTION
**Pull request by abarth**

### CODE REVIEW

File: packages/flutter/lib/src/cupertino/button.dart [hunk 1]
```diff
@@ -5,21 +5,27 @@
 import 'package:flutter/foundation.dart';
 import 'package:flutter/widgets.dart';
 
+// TODO(xster): move this to a common Cupertino color palatte with the next yak.
```
Comment (Hixie) [inline packages/flutter/lib/src/cupertino/button.dart]: palette

File: packages/flutter/lib/src/cupertino/button.dart [hunk 2]
```diff
@@ -35,13 +41,7 @@ const EdgeInsets _kBackgroundButtonPadding =
...
 ///  * <https://developer.apple.com/ios/human-interface-guidelines/ui-controls/buttons/>
 class CupertinoButton extends StatefulWidget {
-  // TODO(xster): move this to a common Cupertino color palatte with the next yak.
-  static const Color kBlue = const Color(0xFF007AFF);
-  static const Color kWhite = const Color(0xFFFFFFFF);
-  static const Color kDisabledBackground = const Color(0xFFA9A9A9);
-  static const Color kDisabledForeground = const Color(0xFFC4C4C4);
-
-
+  /// Creates an iOS style button.
```
Comment (Hixie) [inline packages/flutter/lib/src/cupertino/button.dart]: iOS-style

File: packages/flutter/lib/src/material/colors.dart [hunk 3]
```diff
@@ -73,10 +94,23 @@ class MaterialColor extends ColorSwatch {
 ///
 ///  * [Colors], which defines all of the standard material colors.
 class MaterialAccentColor extends ColorSwatch {
+  /// Creates a color swatch with a variety of shades appropriate for accent
+  /// colors.
```
Comment (Hixie) [inline packages/flutter/lib/src/material/colors.dart]: We should maybe define the term "accent" here somewhere?

File: packages/flutter/lib/src/material/colors.dart [hunk 4]
```diff
@@ -242,6 +274,7 @@ class Colors {
       900: const Color(0xFFB71C1C),
     },
   );
+  static const int _redPrimaryValue = 0xFFF44336;
```
Comment (Hixie) [inline packages/flutter/lib/src/material/colors.dart]: what's the value of moving these?

File: packages/flutter/lib/src/material/colors.dart [hunk 5]
```diff
@@ -267,7 +299,9 @@ class Colors {
       700: const Color(0xFFD50000),
     },
   );
+  static const int _redAccentValue = 0xFFFF5252;
 
+  static const int _pinkPrimaryValue = 0xFFE91E63;
```
Comment (Hixie) [inline packages/flutter/lib/src/material/colors.dart]: this one is presumably in the wrong place

File: packages/flutter/lib/src/material/colors.dart (hunk 4 above)
Comment (abarth) [inline packages/flutter/lib/src/material/colors.dart]: The dartdocs now show up on the web site.

Comment (Hixie) [inline packages/flutter/lib/src/material/colors.dart]: oh never mind i understand

**Comment by Hixie** [comment]
Sir, you are a documentation creation machine.

LGTM.

File: packages/flutter/lib/src/material/colors.dart (hunk 3 above)
Comment (abarth) [inline packages/flutter/lib/src/material/colors.dart]: I linked to the spec, but we might need some more canonical description in ThemeData somewhere to link to.

**Comment by abarth** [comment]
> Sir, you are a documentation creation machine.

I always wanted to write a novel.  I just didn't realize it would be a novel of API docs.

**Pull request merged**